### PR TITLE
refactor: remove Jakarta/main shadow classes

### PR DESCRIPTION
**Pull request by vega113**
Addresses #714

## Summary
- remove the remaining same-path Java shadow copies between `wave/src/main/java` and `wave/src/jakarta-overrides/java`
- keep the Jakarta runtime copies for the former shadowed classes, and keep the main-tree `RobotApiModule.java` while deleting its excluded Jakarta duplicate
- empty the exact duplicate exclude sets in `build.sbt`, update the Jakarta architecture doc, and harden `scripts/jakarta-wrong-edit-guard.sh` for empty `Set()` blocks
- add a regression audit test that fails if same-path duplicates or non-empty exact duplicate exclude sets return

## Verification
- `python3 scripts/assemble-changelog.py --fragments-dir wave/config/changelog.d --output wave/config/changelog.json`
- `python3 scripts/validate-changelog.py --fragments-dir wave/config/changelog.d --changelog wave/config/changelog.json`
- `python3 -m unittest scripts.tests.test_jakarta_wrong_edit_guard scripts.tests.test_issue_714_duplicate_sources -v`
- `sbt test`
- `scripts/jakarta-wrong-edit-guard.sh --base-ref origin/main`
- `git diff --check`

## Notes
- local verification details are recorded in `journal/local-verification/2026-04-12-issue-714-duplicate-source-removal.md`
- `wave/config/changelog.json` remains generated and gitignored; it was assembled locally only to satisfy the SBT resource-generation precondition for verification

## Summary by CodeRabbit

* **Chores**
  * Simplified build behavior: removed curated per-file duplicate-exclude lists so dual-source filtering now relies on directory-level rules.

* **Documentation**
  * Updated architecture docs to describe the simplified dual-source model.
  * Added a cleanup and verification plan with tests to detect remaining duplicate Java sources.

* **Refactor**
  * Removed numerous server-side features and endpoints (robots, search, RPC/WebSocket, auth, attachment/profile/fetch APIs, filters, and startup wiring).

### CODE REVIEW

File: scripts/tests/test_issue_714_duplicate_sources.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      text,
+  )
+  if empty_match:
+    return []
+  match = re.search(
+      rf"val {re.escape(name)}: Set\[String\] = Set\((.*?)\n  \)",
+      text,
+      re.S,
+  )
+  if not match:
```
[the remark this answers is not in the document]
Comment (vega113) [reply scripts/tests/test_issue_714_duplicate_sources.py]: Fixed in 3bfc0454b. The parser no longer depends on "\\n  )"; it matches the declaration flexibly, scans the Set(...) body without indentation assumptions, and fails closed when the declaration is missing or unterminated. Added regression coverage for single-line sets and missing declarations.

File: scripts/tests/test_issue_714_duplicate_sources.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return []
+  return re.findall(r'"([^"]+)"', match.group(1))
```
[the remark this answers is not in the document]
Comment (vega113) [reply scripts/tests/test_issue_714_duplicate_sources.py]: Fixed in 3bfc0454b. parse_set_entries() now handles single-line and multiline Set(...) declarations without relying on a specific closing indentation, and it raises if the declaration cannot be found or closed so the guard cannot silently pass.

[the remark this answers is not in the document]
Comment (vega113) [reply scripts/tests/test_issue_714_duplicate_sources.py]: Fixed in 3bfc0454b. The regression helper now parses the named Set(...) block with whitespace-tolerant matching, scans until the matching close paren, and raises on missing or unterminated declarations. Added focused tests for single-line parsing and fail-closed behavior.

**Comment by vega113** [comment]
Remediation update for PR #858

Worktree: /Users/vega/devroot/worktrees/vega113-incubator-wave-pr-858-monitor
Branch: monitor/vega113-incubator-wave/pr-858
Plan: docs/superpowers/plans/2026-04-12-issue-714-duplicate-source-removal-plan.md

Implementation commits:
- 3bfc0454b test: harden duplicate-source exclude guard

Verification:
- python3 -m unittest scripts.tests.test_issue_714_duplicate_sources ✅
- python3 scripts/assemble-changelog.py && python3 scripts/validate-changelog.py --fragments-dir wave/config/changelog.d --changelog wave/config/changelog.json ✅
- sbt compile ✅
- sbt test ✅ (2251 passed, 0 failed)

Review cleanup:
- Replied to the three open bot review threads on scripts/tests/test_issue_714_duplicate_sources.py
- Resolved all three review threads via GraphQL after pushing 3bfc0454b

Push:
- 3bfc0454bc7ead6f1a5561d6fb269c2c1697e889 -> codex/issue-714-closeout

Remaining blocker:
- GitHub checks are rerunning on the new head; unresolved review threads are at 0.

File: scripts/tests/test_issue_714_duplicate_sources.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+MAIN_ROOT = REPO_ROOT / "wave" / "src" / "main" / "java"
+JAKARTA_ROOT = REPO_ROOT / "wave" / "src" / "jakarta-overrides" / "java"
+BUILD_FILE = REPO_ROOT / "build.sbt"
+
+
+def java_paths(root: Path) -> set[str]:
+  return {
+      str(path.relative_to(root)).replace("\\", "/")
+      for path in root.rglob("*.java")
+  }
```
[the remark this answers is not in the document]
Comment (vega113) [reply scripts/tests/test_issue_714_duplicate_sources.py:18]: Fixed in d7252e199. `java_paths()` now raises when the Java source root is missing, and the regression file includes a focused test so `test_no_same_path_java_duplicates_remain` cannot pass vacuously if either source tree disappears.
